compiler/src/function/main_function.rs

```diff
@@ -61,18 +61,34 @@ impl<'a, F: PrimeField, G: GroupType<F>> ConstrainedProgram<'a, F, G> {
             {
                 let input_variable = input_variable.get().borrow();
                 let name = input_variable.name.name.clone();
-                let input_option = input.get(&name).ok_or_else(|| {
-                    FunctionError::input_not_found(name.clone(), &function.span.clone().unwrap_or_default())
-                })?;
-                let input_value = self.allocate_main_function_input(
-                    cs,
-                    &input_variable.type_.clone(),
-                    &name,
-                    input_option,
-                    &function.span.clone().unwrap_or_default(),
-                )?;
-
-                // Store a new variable for every allocated main function input
+
+                let input_value = match (input.get(&name), input.get_constant(&name)) {
+                    // If input option is found in [main] section.
+                    (Some(input_option), _) => self.parse_constant_main_function_input(
+                        cs,
+                        &input_variable.type_.clone(),
+                        &name,
+                        input_option,
+                        &function.span.clone().unwrap_or_default(),
+                    )?,
+                    // If input option is found in [constants] section.
+                    (_, Some(input_option)) => self.allocate_main_function_input(
```

Comment:
This branch matches on `input.get_constant()`. It should initialize a new constant variable.
```suggestion
                    (_, Some(input_option)) => self.parse_constant_main_function_input(
```